src/Functions/readWkb.cpp

```diff
@@ -0,0 +1,253 @@
+{
+public:
+    explicit FunctionReadWKB() = default;
+
+    static constexpr const char * name = NameHolder::name;
+
+    String getName() const override { return "readWKB"; }
+
+    size_t getNumberOfArguments() const override { return 1; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (checkAndGetDataType<DataTypeString>(arguments[0].get()) == nullptr
+            && checkAndGetDataType<DataTypeFixedString>(arguments[0].get()) == nullptr)
+        {
+            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "First argument should be String");
+        }
+
+        return DataTypeFactory::instance().get(DataTypeName().getName());
+    }
+
+    ColumnPtr
+    executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & /*result_type*/, size_t input_rows_count) const override
+    {
+        if (checkColumn<ColumnString>(*arguments[0].column))
+        {
+            const auto & column_string = checkAndGetColumn<ColumnString>(*arguments[0].column);
+            return executeColumnImpl(column_string, input_rows_count);
+        }
+        else
+        {
+            const auto & column_string = checkAndGetColumn<ColumnFixedString>(*arguments[0].column);
+            return executeColumnImpl(column_string, input_rows_count);
+        }
+    }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionReadWKB<DataTypeName, Geometry, Serializer, NameHolder>>(); }
+
+private:
+    template <typename T>
+    ColumnPtr executeColumnImpl(const T & column, size_t input_rows_count) const
+    {
+        Serializer serializer;
+        for (size_t i = 0; i < input_rows_count; ++i)
+        {
+            auto str = column.getDataAt(i).toString();
+            ReadBuffer in_buffer(str.data(), str.size(), 0);
+
+            auto object = parseWKBFormat(in_buffer);
+            auto boost_object = std::get<Geometry>(object);
+            serializer.add(boost_object);
+        }
+
+        return serializer.finalize();
+    }
+};
+
+struct ReadWKBPointNameHolder
+{
+    static constexpr const char * name = "readWKBPoint";
+};
+
+struct ReadWKBLineStringNameHolder
+{
+    static constexpr const char * name = "readWKBLineString";
+};
+
+struct ReadWKBMultiLineStringNameHolder
+{
+    static constexpr const char * name = "readWKBMultiLineString";
+};
+
+struct ReadWKBPolygonNameHolder
+{
+    static constexpr const char * name = "readWKBPolygon";
+};
+
+struct ReadWKBMultiPolygonNameHolder
+{
+    static constexpr const char * name = "readWKBMultiPolygon";
+};
+
+}
+
+REGISTER_FUNCTION(ReadWKB)
+{
+    factory.registerFunction<FunctionReadWKB<DataTypePointName, CartesianPoint, PointSerializer<CartesianPoint>, ReadWKBPointNameHolder>>(
```

Comment:
I'd prefer if we use the standarized function names directly, and use aliases if we don't like the names:
* ST_PointFromWKB
* ST_LineFromWKB
* ST_MLineFromWKB
* ST_PolyFromWKB
* ST_MPolyFromWKB